graph/telemetry/istio/istio.go

```diff
@@ -551,7 +564,23 @@ func addNode(trafficMap graph.TrafficMap, cluster, serviceNs, service, workloadN
 }
 
 func timeSeriesHash(cluster, serviceNs, service, workloadNs, workload, app, version string) string {
-	return fmt.Sprintf("%x", sha256.Sum256([]byte(strings.Join([]string{cluster, serviceNs, service, workloadNs, workload, app, version}, ":"))))
+	var b strings.Builder
+	b.WriteString(cluster)
+	b.WriteByte(':')
+	b.WriteString(serviceNs)
+	b.WriteByte(':')
+	b.WriteString(service)
+	b.WriteByte(':')
+	b.WriteString(workloadNs)
+	b.WriteByte(':')
+	b.WriteString(workload)
+	b.WriteByte(':')
+	b.WriteString(app)
+	b.WriteByte(':')
+	b.WriteString(version)
+
+	sum := sha256.Sum256([]byte(b.String()))
+	return hex.EncodeToString(sum[:])
```

Comment:
Using strings.Builder is actually slower that strings.Join, and more verbose. See my analysis in https://redhat-internal.slack.com/archives/CHR7P722J/p1744746871944699.

The savings here could be in the removal of `Sprintf`.